gen/one_hot_gen.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// one_hot_gen.dart
+// Generate one_hot codecs.
+//
+// 2023 May 09
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+//
+
+import 'dart:io';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/one_hot.dart';
+import 'package:rohd_hcl/src/utils.dart';
+
+Future<void> oneHotGen() async {
+  const pos = 8;
+  final w = log2Ceil(pos + 1);
+  final mod = BinaryToOneHot(Const(pos, width: w));
+  await mod.build();
+  final res = mod.generateSynth();
+  File('build/${mod.definitionName}.v').openWrite().write(res);
+
+  final val2 = BigInt.from(2).pow(pos);
+  final mod2 = OneHotToBinary(Const(val2, width: pos + 1));
```

Comment:
Do we have to pass constants in or can we pass `Logic`s in?  Does a fixed-value help the schematic view in some way?